docs/deploy.md

```diff
@@ -9,22 +9,38 @@ The following are the parameters supported by this goal in addition to the [comm
 
-When determining which resolved dependencies to copy for the `copyDependencies` configuration, only scopes compile, runtime and system are included. This ensures provided scope and test scope dependencies are not copied. Also, the `type` is defaulted to `jar`. If your scenario is more complex, consider using the `copy` or `copy-dependencies` goal in the `maven-dependency-plugin` instead.
+The `dependencyGroup` parameter within the `copyDependencies` can contain the following parameters.
+
+| Parameter | Description | Required |
+| --------  | ----------- | -------  |
+| dependency | A collection of `dependency` parameters that specify the coordinate of the Maven dependency to copy. | Yes |
+| location | The optional directory to which the dependencies are copied. This can be an absolute path, or a relative path to the target server configuration directory. If not specified, the `location` from the `copyDependencies` is used.| No |
+
+The `dependency` parameter within the `copyDependencies` or `dependencyGroup` can contain the following parameters.
+
+| Parameter | Description | Required |
+| --------  | ----------- | -------  |
+| groupId | The groupId of the Maven dependency to be copied. The `artifactId` and `version` are optional. If only `groupId` is specified, all resolved dependencies with a matching `groupId` are copied to the specified or default location along with their transitive dependencies. | Yes |
+| artifactId | The artifactId of the Maven dependency to be copied. If an `artifactId` is specified, the resolved dependency with a matching `groupId` and `artifactId` is copied to the specified or default location along with its transitive dependencies. The `artifactId` may also end with a `*` to match all artifacts that have an `artifactId` that start with the specified string. | No |
+| version | The version of the Maven dependency to be copied. You must specify the `version` for any dependency that is not configured in the Maven `dependencies` or Maven `dependencyManagement` section of the `pom.xml` file. | No |
+
+When determining which resolved dependencies to copy for the `copyDependencies` configuration, only scopes compile, runtime, system and provided are included. This ensures test scope dependencies are not copied. Please note that dependencies with scope compile, runtime, or system will still be packaged within the application unless configured otherwise. If you do not want the dependency within the application, then consider removing the dependency from the Maven `dependencies` or Maven `dependencyManagement` section of the pom.xml and specify the full coordinate with `version` within a `dependency` in `copyDependencies`. Alternatively, you change the dependency scope to provided. The `type` is also defaulted to `jar`. If your scenario is more complex, consider using the `copy` or `copy-dependencies` goal in the `maven-dependency-plugin` instead.
```

Comment:
```suggestion
When determining which resolved dependencies to copy for the `copyDependencies` configuration, only scopes compile, runtime, system and provided are included. This ensures test scope dependencies are not copied. Please note that dependencies with scope compile, runtime, or system will still be packaged within the application unless configured otherwise. If you do not want the dependency within the application, then consider removing the dependency from the Maven `dependencies` or Maven `dependencyManagement` section of the pom.xml and specify the full coordinate with `version` within a `dependency` in `copyDependencies`. Alternatively, you could change the dependency scope to provided. The `type` is also defaulted to `jar`.  A dependency that is only configured in `copyDependencies` but not in the Maven `dependencies` or Maven `dependencyManagement` will be treated as a 'provided'-scoped dependency in calculating transitive dependencies.  If your scenario is more complex, consider using the `copy` or `copy-dependencies` goal in the `maven-dependency-plugin` instead.
```

Reply:
I just updated the doc, but made a slight tweak to what you suggested. Basically, if they specify the `version` in `copyDependencies`, then it will be treated as provided scope.

Reply:
Not sure what we did wrong but this all looks duplicated now ... 

Also since we have to fix anyway, I like "Alternatively, you could change"  over "Alternatively, you change".

Reply:
That was my fault. I copied your paragraph in to compare the original paragraph with it, then forgot to delete the original paragraph. See if what I just pushed looks right.

Reply:
We're getting close...let me just probe on one special case:

So if I have a compile-scoped dependency with the version in `<dependencies>`  and I list the full G:A:V in 'copyDependencies'... it gets treated as provided ?    I'm a bit surprised we wouldn't find the dependency and honor its existing scope.

Reply:
Correct, we said in the original design and proposal that if they specified the version in the `copyDependencies` we would trust the user and go resolve that artifact instead of what was specified (and already resolved) in the Maven dependencies. The code that finds/resolves an artifact always uses provided scope. There could probably be an argument to honor the scope but not the version of the Maven dependency in that scenario, but I think we can let the user drive that and let us know?